ncm-grub/src/main/perl/grub.pm

```diff
@@ -191,11 +193,25 @@ sub password
     my $val = $tree->{option} ? "--$tree->{option} " : "";
     $val .= $password;
 
-    $grub_fh->add_or_replace_lines(qr/^password\s+/,
-                                   qr/^password $val$/,
-                                   "password $val\n",
-                                   $self->main_section_offset($grub_fh),
-                                   );
+    if ( -d $GRUB2_DIR ) {
```

Comment:
Good point about grub2 x86_64 UEFI using /boot/efi/EIF/redhat/grub.cfg. Am I right that we do not support any of the other architectures?

But grub2-efi depends on grub2-tools, which does provide this directory.

# rpm --requires -qip /root/grub2-efi-2.02-0.44.el7.x86_64.rpm
grub2-tools = 1:2.02-0.44.el7

# rpm -ql grub2-tools|grep /boot/grub2
/boot/grub2

The key part is /etc/grub.d/01_users, which is also in grub2-tools.

[root@izvmn08azgh7 tmp]# rpm -ql grub2-tools| grep users
/etc/grub.d/01_users
[root@izvmn08azgh7 tmp]# cat /etc/grub.d/01_users
#!/bin/sh -e
cat << EOF
if [ -f \${prefix}/user.cfg ]; then
  source \${prefix}/user.cfg
  if [ -n "\${GRUB2_PASSWORD}" ]; then
    set superusers="root"
    export superusers
    password_pbkdf2 root \${GRUB2_PASSWORD}
  fi
fi
EOF

It sources the user.cfg file, meaning you don't have to regenerate the grub config. I think it should work for both UEFI and BIOS.